### PR TITLE
RHBPMS-1698 -  Build & Deploy fails due to duplicate ProcessID after copying business process (#274)

### PR DESCRIPTION
**Pull request by tsurdilo**

### CODE REVIEW

**Comment by jlindop** [comment]
jenkins retest this

File: jbpm-designer-backend/src/main/java/org/jbpm/designer/server/BusinessProcessCopyHelper.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+/**
+ * CopyHelper for Business Processes
+ */
+@ApplicationScoped
+public class BusinessProcessCopyHelper implements CopyHelper {
+
+    private IOService ioService;
+    private CommentedOptionFactory commentedOptionFactory;
+    private Bpmn2TypeDefinition bpmn2ResourceType;
+    private IDiagramProfileService profileService;
```
Comment (jlindop) [inline jbpm-designer-backend/src/main/java/org/jbpm/designer/server/BusinessProcessCopyHelper.java]: This member isn't used.

**Comment by jlindop** [comment]
@tsurdilo  It might be worth changing the BP Name as well as the ID when copying. When you build&deploy a project with a copied process, both processes are listed with the same name in the "Process Definitions" table.

**Comment by jlindop** [comment]
@tsurdilo  I tested this with kie-wb-6.4.1-SNAPSHOT-eap6_4.war built from kie-wb-distributions/kie-wb and there's a startup error:
14:08:35,527 ERROR [org.jboss.msc.service.fail] (MSC service thread 1-2) MSC000001: Failed to start service jboss.deployment.unit."kie-wb-6.4.1-SNAPSHOT-eap6_4.war".WeldStartService: org.jboss.msc.service.StartException in service jboss.deployment.unit."kie-wb-6.4.1-SNAPSHOT-eap6_4.war".WeldStartService: Failed to start service
	at org.jboss.msc.service.ServiceControllerImpl$StartTask.run(ServiceControllerImpl.java:1936) [jboss-msc-1.1.5.Final-redhat-1.jar:1.1.5.Final-redhat-1]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [rt.jar:1.8.0_65]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [rt.jar:1.8.0_65]
	at java.lang.Thread.run(Thread.java:745) [rt.jar:1.8.0_65]
Caused by: org.jboss.weld.exceptions.UnproxyableResolutionException: WELD-001435 Normal scoped bean class org.jbpm.designer.server.BusinessProcessCopyHelper is not proxyable because it has no no-args constructor - Managed Bean [class org.jbpm.designer.server.BusinessProcessCopyHelper] with qualifiers [@Default @Any].

I think the best way to fix this is to remove the constructor and inject the constructor's parameters directly, i.e.
    @Inject
    @Named("ioStrategy")
    private IOService ioService;
    @Inject
    private CommentedOptionFactory commentedOptionFactory;
    @Inject
    private Bpmn2TypeDefinition bpmn2ResourceType;
However, I tried this, and immediately after copying a BP, EAP's log shows this:
15:17:12,887 INFO  [org.guvnor.common.services.builder.ResourceChangeIncrementalBuilder] (EJB default - 8) Incremental build request being processed: default://master@uf-playground/K/src/main/resources/l2.bpmn2 (added).
15:17:12,935 ERROR [org.drools.core.xml.ExtensibleXmlParser] (EJB default - 8) (null: 2, 687): cvc-elt.1: Cannot find the declaration of element 'bpmn2:definitions'.
15:17:12,936 ERROR [org.drools.core.xml.ExtensibleXmlParser] (EJB default - 8) (null: 42, 63): cvc-elt.4.2: Cannot resolve 'dc:Point' to a type definition for element 'di:waypoint'.
15:17:12,937 ERROR [org.drools.core.xml.ExtensibleXmlParser] (EJB default - 8) (null: 43, 63): cvc-elt.4.2: Cannot resolve 'dc:Point' to a type definition for element 'di:waypoint'.
15:17:12,937 ERROR [org.drools.core.xml.ExtensibleXmlParser] (EJB default - 8) (null: 49, 63): cvc-elt.4.2: Cannot resolve 'dc:Point' to a type definition for element 'di:waypoint'.
15:17:12,937 ERROR [org.drools.core.xml.ExtensibleXmlParser] (EJB default - 8) (null: 50, 63): cvc-elt.4.2: Cannot resolve 'dc:Point' to a type definition for element 'di:waypoint'.
15:17:12,938 ERROR [org.drools.core.xml.ExtensibleXmlParser] (EJB default - 8) (null: 57, 90): cvc-elt.4.2: Cannot resolve 'bpsim:Scenario' to a type definition for element 'bpsim:Scenario'.
15:17:12,938 ERROR [org.drools.core.xml.ExtensibleXmlParser] (EJB default - 8) (null: 58, 93): cvc-elt.4.2: Cannot resolve 'bpsim:ScenarioParameters' to a type definition for element 'bpsim:ScenarioParameters'.
15:17:12,938 ERROR [org.drools.core.xml.ExtensibleXmlParser] (EJB default - 8) (null: 59, 151): cvc-elt.4.2: Cannot resolve 'bpsim:ElementParameters' to a type definition for element 'bpsim:ElementParameters'.
15:17:12,938 ERROR [org.drools.core.xml.ExtensibleXmlParser] (EJB default - 8) (null: 60, 67): cvc-elt.4.2: Cannot resolve 'bpsim:TimeParameters' to a type definition for element 'bpsim:TimeParameters'.
15:17:12,938 ERROR [org.drools.core.xml.ExtensibleXmlParser] (EJB default - 8) (null: 61, 64): cvc-elt.4.2: Cannot resolve 'bpsim:Parameter' to a type definition for element 'bpsim:ProcessingTime'.
15:17:12,939 ERROR [org.drools.core.xml.ExtensibleXmlParser] (EJB default - 8) (null: 66, 151): cvc-elt.4.2: Cannot resolve 'bpsim:ElementParameters' to a type definition for element 'bpsim:ElementParameters'.
15:17:12,939 ERROR [org.drools.core.xml.ExtensibleXmlParser] (EJB default - 8) (null: 67, 73): cvc-elt.4.2: Cannot resolve 'bpsim:ControlParameters' to a type definition for element 'bpsim:ControlParameters'.

**Comment by tsurdilo** [comment]
@jlindop the error is because i did not cherry-pick https://github.com/droolsjbpm/jbpm-designer/commit/4a42dc19356b9cf31a72f06d9e2bfe12953d594f

will do now

**Comment by tsurdilo** [comment]
@jlindop i will redo this PR to include both commits that are needed

**Pull request closed without merging**